dashboard/app/models/course_offering.rb

```diff
@@ -385,6 +386,15 @@ def self.properties_from_file(content)
   # seed_all
   def self.seed_record(file_path)
     properties = properties_from_file(File.read(file_path))
+
+    key = properties[:self_paced_pl_course_offering_key]
+    self_paced_pl_course_offering_key = CourseOffering.find_by_key(key)
+    if self_paced_pl_course_offering_key.nil? && !key.nil?
+      warn "self_paced_pl_course_offering_key: #{key} not found. Please seed again to fix."
+    else
+      properties[:self_paced_pl_course_offering_id] = self_paced_pl_course_offering_key&.id
```

Comment:
looks better! one naming suggestion:
```suggestion
    new_self_paced_pl_course_offering = CourseOffering.find_by_key(key)
    if new_self_paced_pl_course_offering.nil? && !key.nil?
      warn "self_paced_pl_course_offering_key: #{key} not found. Please seed again to fix."
    else
      properties[:self_paced_pl_course_offering_id] = new_self_paced_pl_course_offering&.id
```